tests/test_evaluation.py

```diff
@@ -513,27 +513,64 @@ def mock_expr_tree():
     return tree
 
 
-def test_eval_expr_3_good(mock_expr_tree):
+def test_eval_expr_3_left_good(mock_left_expr_tree):
     activation = Mock()
     evaluator = Evaluator(
-        mock_expr_tree,
+        mock_left_expr_tree,
         activation
     )
     assert evaluator.evaluate() == celtypes.IntType(6)
 
 
-def test_eval_expr_3_bad_override(mock_expr_tree):
+def test_eval_expr_3_bad_override(mock_left_expr_tree):
     def bad_condition(a, b, c):
         raise TypeError
     activation = Mock()
     evaluator = Evaluator(
-        mock_expr_tree,
+        mock_left_expr_tree,
         activation,
         functions={"_?_:_": bad_condition}
     )
     with raises(celpy.evaluation.CELEvalError):
         evaluator.evaluate()
 
+@fixture
+def mock_right_expr_tree():
```

Comment:
Should there be a test for some conditionalor such that in `false ? f1() : f2()` the function `f1` is never evaluated?

Reply:
Yes, that should be `def test_eval_expr_3_left_good(mock_left_expr_tree)`, later in the module.

Reply:
Ok, maybe I don't understand, but my initial assumption was that `test_eval_expr_3_right_good` only checks if the evaluation of the conditionalor results in the right value, but not necessarily that the other wasn't also evaluated. My concern was that if there was a regression, like your changes to `evaluation.py` were accidentally reverted, the test cases would not notice the evaluation of extraneous branches in the conditionalor.